src/Clock.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+
+type Props = {
+  name: string;
+};
+
+type State = {
+  date: Date;
+};
+
+export class Clock extends React.PureComponent<Props, State> {
+  state: State = {
+    date: new Date(),
+  };
+
+  timerId = 0;
+
+  componentDidMount(): void {
+    this.timerId = window.setInterval(() => {
+      this.setState({ date: new Date() });
+      // eslint-disable-next-line no-console
+      console.info(this.state.date.toUTCString().slice(-12, -4));
+    }, 1000);
+  }
+
+  componentDidUpdate(
+    prevProps: Readonly<Props>,
+  ): void {
+    if (prevProps.name !== this.props.name) {
+      // eslint-disable-next-line no-console
+      console.debug(`Renamed from ${prevProps.name} to ${this.props.name}`);
+    }
+  }
+
+  componentWillUnmount(): void {
+    window.clearInterval(this.timerId);
+  }
+
+  render(): React.ReactNode {
+    const { date } = this.state;
+    const { name } = this.props;
+
+    return (
+      <div className="Clock">
+        <strong className="Clock__name">
+          {name}
+        </strong>
+
+        {' time is '}
+
+        <span className="Clock__time">
+          {date.toUTCString().slice(-12, -4)}
```

Comment:
You do a similar operation twice, so it's worth taking it out as a separate helper.